src/modals/SettingsModal/NetworkSettingsSection.tsx

```diff
@@ -72,8 +72,10 @@ const NetworkSettingsSection = () => {
   }
 
   const handleNetworkPresetChange = useCallback(
-    (option: typeof networkSelectOptions[number] | undefined) => {
-      if (option && option.value !== selectedNetwork) {
+    (option?: SelectOption<NetworkName>) => {
+      if (option === undefined) return
+
+      if (option.value !== selectedNetwork) {
```

Comment:
Why splitting it in 2 conditionals and not just one?
```ts
if (option === undefined || option.value === selectedNetwork) return
```

Reply:
Was focused on satisfying the typechecker - yeah one is better